contracts/peripheral/ImportProxy.sol

```diff
@@ -0,0 +1,259 @@
+    IWeth public immutable weth;
+    IERC20 public immutable dai;
+    IGemJoin public immutable wethJoin;
+    IDaiJoin public immutable daiJoin;
+    IController public immutable controller;
+    address public immutable treasury;
+    IImportProxy public immutable splitter;
+    IProxyRegistry public immutable proxyRegistry;
+
+    bytes32 public constant WETH = "ETH-A";
+    bool public constant MTY = true;
+
+    constructor(IController controller_, IPool[] memory pools_, IProxyRegistry proxyRegistry_) public {
+        ITreasury _treasury = controller_.treasury();
+
+        weth = _treasury.weth();
+        dai = _treasury.dai();
+        daiJoin = _treasury.daiJoin();
+        wethJoin = _treasury.wethJoin();
+        vat = _treasury.vat();
+
+        controller = controller_;
+        treasury = address(_treasury);
+        splitter = IImportProxy(address(this)); // This contract has two functions, as itself, and delegatecalled by a dsproxy.
+
+        proxyRegistry = proxyRegistry_;
+
+        // Allow pool to take fyDai for trading
+        for (uint i = 0 ; i < pools_.length; i++) {
+            pools_[i].fyDai().approve(address(pools_[i]), type(uint256).max);
+        }
+
+        // Allow treasury to take weth for posting
+        _treasury.weth().approve(address(_treasury), type(uint256).max);
+
+        // Allow wethJoin to move weth out of vat for this proxy
+        _treasury.vat().hope(address(_treasury.wethJoin()));
+
+        // Allow daiJoin to take Dai for paying debt
+        _treasury.dai().approve(address(_treasury.daiJoin()), type(uint256).max);
+    }
+
+    /// --------------------------------------------------
+    /// ImportProxy via dsproxy: Fork and Split
+    /// --------------------------------------------------
+
+    /// @dev Fork part of a user MakerDAO vault to ImportProxy, and call splitter to transform it into a Yield vault
+    /// This function can be called from a dsproxy that already has a `vat.hope` on the user's MakerDAO Vault
+    /// @param pool fyDai Pool to use for migration, determining maturity of the Yield Vault
+    /// @param user CDP Vault to import
+    /// @param wethAmount Weth collateral to import
+    /// @param debtAmount Normalized debt to move ndai * rate = dai
+    function importPosition(IPool pool, address user, uint256 wethAmount, uint256 debtAmount) public {
+        require(user == msg.sender || proxyRegistry.proxies(user) == msg.sender, "Restricted to user or its dsproxy"); // Redundant?
+        splitter.hope(msg.sender);                     // Allow the user or proxy to give splitter the MakerDAO vault.
+        vat.fork(                                      // Take the treasury vault
+            WETH,
+            user,
+            address(splitter),
+            wethAmount.toInt256(),
+            debtAmount.toInt256()
+        );
+        splitter.nope(msg.sender);                     // Disallow the user or proxy to give splitter the MakerDAO vault. Needed?
```

Comment:
I think it's worth keeping this for our peace of mind. If we didn't revoke it, anybody could call `importPosition` on any user that previously imported their position and brought it back to Maker.

Reply:
Agree